### PR TITLE
Add Windows Server 2022 based images

### LINKED ISSUE #17656: Publish Windows Server 2022 Envoy images
Windows Server 2022 comes with many new features but the most notable one is support for `SO_ORIGINAL_DST` and traffic redirection via HNS policies. These features are an absolute requirement to be able to run Envoy as a sidecar in a service mesh solution.

We should publish windows server 2022 images of Envoy (**ideally** based on Nanoserver)

Limitations:

1. windows-server-latest CI is not compatible with window server 2022 which means that we can't build the image from a windows host
2. `docker buildx` is not compatible with `RUN` commands which we use to set the path.

Problem (1) is not possible to solve, so we will have to solve (2).

We use `RUN` commands for two reasons:

1. Setting up the Envoy user, which can be moved into a runtime script.
2. Place `envoy.exe` into a location that is already in the path and thus remove the need for running `setx` commands.

tl;dr: We should build Windows images from linux with `docker buildx`

### PR DESCRIPTION
**Pull request by davinci26**
Signed-off-by: Sotiris Nanopoulos <sonanopo@microsoft.com>

Commit Message:

Publishes Windows images based on nanoserver 2022. Fixes #17656

Additional Description:

To motivate some of the choices here:
1. Nanoserver was chosen instead of servercore because of the size reduction.
2. New job was added since we cant build these images on the OS of the previous job. Neither can we build the old images on windows server 2022

Risk Level:
Low, CI only

Testing: through CI
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by davinci26** [comment]
@phlax @wrowe can you please review

cc @envoyproxy/windows-dev

**Comment by wrowe** [comment]
/wait

Rechecking CNCF guidance.

File: .azure-pipelines/pipelines.yml
Comment (wrowe) [inline .azure-pipelines/pipelines.yml:582]: If we are distinguishing WindowsServer and WindowsNanoServer, shouldn't this become part of the job name? Isn't it reasonable to expect that both for 2022 might be interesting to one user or another, until we have distroless to let the user choose?

Comment (davinci26) [reply .azure-pipelines/pipelines.yml:582]: > If we are distinguishing WindowsServer and WindowsNanoServer, shouldn't this become part of the job name? 

Yes I can change that

> Isn't it reasonable to expect that both for 2022 might be interesting to one user or another, until we have distroless to let the user choose?

I dont see the motivation for servercore given the size and that nanoserver is the blessed way forward. I am having a lazy approach where we could add servercore images if someone asks for it. I kept the servercore based images for 2019 only because we already publish it

Comment (davinci26) [reply .azure-pipelines/pipelines.yml]: fixed

**Review by wrowe — APPROVED** [review]
Looks good with the changes on naming, thx!

**Comment by davinci26** [comment]
The windows-docker step did not hang, its just that it exists on main and not in my pipeline. I replaced it with the new pipelines

**Comment by davinci26** [comment]
@wrowe pipeline is clean now although it probably requires an approval from @envoyproxy/senior-maintainers to get this one merged since with the pipeline change the old job shows as expected

**Comment by davinci26** [comment]
@lizan since you are on-call this week can you take a look at this PR? 

There is an issue because the CI jobs got modified, the job that got deleted (windows docker) is marked as expected by the CI. So the CI will never become fully green.

File: .azure-pipelines/pipelines.yml
Comment (lizan) [inline .azure-pipelines/pipelines.yml:583]: can we use a build matrix for this instead of duplicating?

Comment (davinci26) [reply .azure-pipelines/pipelines.yml:583]: There are multiple reasons we can't use buildx to do this:

1. buildx is not supported on Windows
2. We rely on RUN commands on the Dockerfile which prohibits us from running buildx on linux. See https://github.com/envoyproxy/envoy/issues/17656
3. Building with buildx on linux would push windows foreign layers on our dockerhub and this would violate the license of the base layer.

Also Windows server images on ci do not come with hyper-v enabled so have to use two different machines for the two different versions of the OS.

File: .azure-pipelines/pipelines.yml
Comment (wrowe) [inline .azure-pipelines/pipelines.yml:550]: Did you break the indention here?

Comment (davinci26) [reply .azure-pipelines/pipelines.yml]: yeah I did

**Review by mattklein123 — APPROVED** [review]

**Comment by davinci26** [comment]
@mattklein123 PTAL, the CI can never go green because I replaced the CI job that is marked as pending with a different one

File: .azure-pipelines/pipelines.yml
Comment (davinci26) [reply .azure-pipelines/pipelines.yml]: Resolved, thanks for the pointer offline

File: .azure-pipelines/pipelines.yml
Comment (lizan) [inline .azure-pipelines/pipelines.yml:554]: It might be better to use `windows-2019` here so it is not affected when azp update its latest image.

**Review by lizan — COMMENTED** [review]
Also add the image to https://github.com/envoyproxy/envoy/blob/main/docs/root/start/install.rst#pre-built-envoy-docker-images

**Pull request merged**
